packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -365,11 +378,12 @@ class IOSDevice extends Device {
   String get _buildVersion => _getDeviceInfo(id, 'BuildVersion');
 
   @override
-  DeviceLogReader get logReader {
-    if (_logReader == null)
-      _logReader = new _IOSDeviceLogReader(this);
+  DeviceLogReader get logReader => logReaderForApp(null);
 
-    return _logReader;
+  @override
+  DeviceLogReader logReaderForApp(ApplicationPackage app) {
+    _logReaders ??= <ApplicationPackage, _IOSDeviceLogReader>{};
```

Comment:
I was purposely letting them live as long as the process to match what was being done before with the instance variable.  Yes, this could *theoretically* grow without bound, but practically speaking, we're only going to have one `ApplicationPackage` we're passing around.

If we felt like being super careful, we could override `equals()` and `hashCode()` in `ApplicationPackage`...